public/app/features/explore/state/query.ts

```diff
@@ -531,7 +536,15 @@ export const runQueries = createAsyncThunk<void, RunQueriesOptions>(
     if (cachedValue) {
       newQuerySource = combineLatest([of(cachedValue), correlations$]).pipe(
         mergeMap(([data, correlations]) =>
-          decorateData(data, queryResponse, absoluteRange, refreshInterval, queries, correlations)
+          decorateData(
+            data,
+            queryResponse,
+            absoluteRange,
+            refreshInterval,
+            queries,
+            correlations,
+            isCorrelationEditorMode && isLeftPane
```

Comment:
I think it could be a bit more readable if we used `isCorrelationSource` rather than `isLeftPane`. Same in components/selectors. For someone reading the code, it might not be obvious at first sight why we have logic specific to the left pane only.

Reply:
I hear what you're saying, but I do think keeping this variable named what it is we're actually evaluating without feature specific context might be more valuable. We already have logic in the toolbar around left/right panes, and the `[0][0]` evaluation is strange enough for people not familiar with how the panes are returned that elaborating the variable name on what it actually means on the screen seems useful.

That said, I can absolutely change this if you think it's best.

Reply:
>  (...) keeping this variable named what it is we're actually evaluating without feature specific context might be more valuable (...)

It's interesting, I actually prefer to have variable names containing _feature specific context_, but it's a minor thing. I think you can leave it as it is.